File: VB/WPFDataGridApp15/ViewModel.vb
Imports System
Imports System.Collections.ObjectModel
Imports DevExpress.Xpf.Core.Commands

Namespace WPFDataGridApp15

    Public Class ViewModel

        Private _Data As ObservableCollection(Of WPFDataGridApp15.DataClass), _ExpandedItems As ObservableCollection(Of WPFDataGridApp15.DataClass), _CollapseAllCommand As DelegateCommand(Of Object)

        Public Property Data As ObservableCollection(Of DataClass)
            Get
                Return _Data
            End Get

            Private Set(ByVal value As ObservableCollection(Of DataClass))
                _Data = value
            End Set
        End Property

        Public Property ExpandedItems As ObservableCollection(Of DataClass)
            Get
                Return _ExpandedItems
            End Get

            Private Set(ByVal value As ObservableCollection(Of DataClass))
                _ExpandedItems = value
            End Set
        End Property

        Public Property CollapseAllCommand As DelegateCommand(Of Object)
            Get
                Return _CollapseAllCommand
            End Get

            Private Set(ByVal value As DelegateCommand(Of Object))
                _CollapseAllCommand = value
            End Set
        End Property

        Public Sub New()
            Data = New ObservableCollection(Of DataClass)()
            ExpandedItems = New ObservableCollection(Of DataClass)()
            CollapseAllCommand = New DelegateCommand(Of Object)(New Action(Of Object)(AddressOf CollapseAllExecute))
            GenerateData(20)
            ExpandedItems.Add(Data(1))
            ExpandedItems.Add(Data(10))
        End Sub

        Private Sub CollapseAllExecute(ByVal arg As Object)
            ExpandedItems.Clear()
        End Sub

        Private Sub GenerateData(ByVal count As Integer)
            For i As Integer = 0 To count - 1
                Data.Add(New DataClass(i))
            Next
        End Sub
    End Class
End Namespace

File: CS/WPFDataGridApp15/ViewModel.cs
using System;
using System.Collections.ObjectModel;
using DevExpress.Xpf.Core.Commands;

namespace WPFDataGridApp15
{
	public class ViewModel
	{
		public ObservableCollection<DataClass> Data { get; private set; }
		public ObservableCollection<DataClass> ExpandedItems { get; private set; }

		public DelegateCommand<object> CollapseAllCommand { get; private set; }

		public ViewModel()
		{
			Data = new ObservableCollection<DataClass>();
			ExpandedItems = new ObservableCollection<DataClass>();
			CollapseAllCommand = new DelegateCommand<object>(new Action<object>(CollapseAllExecute));
			GenerateData(20);

			ExpandedItems.Add(Data[1]);
			ExpandedItems.Add(Data[10]);
		}

		private void CollapseAllExecute(object arg)
		{
			ExpandedItems.Clear();
		}

		private void GenerateData(int count)
		{
			for ( int i = 0; i < count; i++ )
				Data.Add(new DataClass(i));
		}
	}
}

